Introduce horizontally scrollable segmented control setting

RasmusKjeldgaard opened the pull request:
## Which issue does this PR close?

This PR closes #2277 

## What is the new behavior?
There is now an input (`scrollable`) on Segmented Control that can be set to true, to add `overflow-x: scroll` to the segmented control as a whole. This way, if the Segmented Control is constrained in its width, it will get a scroll bar horizontally, to reveal the overflowed segments. 

Setting this input also sets `scrollable` on the underlying `ion-segment`, which makes sure it does not fire a click event when the segments are scrolled/swiped, but only when they are directly clicked. 

Furthermore, a CSS custom property is added to control the overflow behavior in situations when the segmented control itself should not be scrollable, but where it participates in a scrollable context.  

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to `stable` via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

RasmusKjeldgaard committed:
✨ Add scrollable input to SegmentedControl

RasmusKjeldgaard committed:
📝  Add scrollable documentation entry

RasmusKjeldgaard committed:
💄 Hide scrollbar on mobile devices

RasmusKjeldgaard committed:
📝  Add documentation for new scrollable setting

RasmusKjeldgaard committed:
🐛  Make items public

MadsBuchmann reviewed (commented):
As you've mentioned yourself there's a rogue vertical spacebar whenever you force scrollbars 🥷  
<img width="616" alt="Screenshot 2022-05-19 at 14 32 45" src="https://user-images.githubusercontent.com/17269698/169293964-d01831df-efd6-484c-b43c-58729568087f.png">

Other than that it looks good imo! 

However, why is the overflow-x custom css property only available when we have scrollable applied? Couldn't we might as well make it usable at all times for simplicity?

RasmusKjeldgaard committed:
✨ Add CSS Custom Property for overflow-y

This should make the solution a bit more robust as project have better
control in advanced scenarios

RasmusKjeldgaard commented:
Closing this as it will need some more design work. 
Replaced by #2237
